CMakeLists.txt

```
@@ -3111,6 +3112,20 @@ if (SDL_ASAN)
   endif()
 endif()
 
+if (SDL_CLANG_TIDY)
+    cmake_minimum_required(VERSION 3.6)
+    find_program(CLANG_TIDY clang-tidy)
+
+    if(CLANG_TIDY MATCHES "CLANG_TIDY-NOTFOUND")
+        set(HAVE_CLANG_TIDY OFF)
+    else()
+        set(HAVE_CLANG_TIDY ON)
+        set(CMAKE_C_CLANG_TIDY ${CLANG_TIDY})
+        set(CMAKE_CXX_CLANG_TIDY ${CLANG_TIDY})
+        set(CMAKE_OBJC_CLANG_TIDY ${CLANG_TIDY})
+    endif()
```

Variables ending with `-NOTFOUND` are [treated as false](https://cmake.org/cmake/help/latest/command/if.html#basic-expressions).
I also inverted the conditional (this is a personal preference)
```suggestion
    if(CLANG_TIDY)
        set(HAVE_CLANG_TIDY ON)
        set(CMAKE_C_CLANG_TIDY ${CLANG_TIDY})
        set(CMAKE_CXX_CLANG_TIDY ${CLANG_TIDY})
        set(CMAKE_OBJC_CLANG_TIDY ${CLANG_TIDY})
    else()
        set(HAVE_CLANG_TIDY OFF)
    endif()
```

---

Switched it around, wasn't aware of the `-NOTFOUND` special case.